front_end/panels/react_devtools/ReactDevToolsModel.ts

```diff
@@ -31,91 +30,163 @@ type ReactDevToolsBindingsBackendExecutionContextUnavailableEvent = Common.Event
+    window.addEventListener('beforeunload', () => this.#bridge?.shutdown());
+  }
 
-    rdtBindingsModel.addEventListener(ReactNativeModels.ReactDevToolsBindingsModel.Events.BackendExecutionContextCreated, this.onBackendExecutionContextCreated, this);
-    rdtBindingsModel.addEventListener(ReactNativeModels.ReactDevToolsBindingsModel.Events.BackendExecutionContextUnavailable, this.onBackendExecutionContextUnavailable, this);
-    rdtBindingsModel.addEventListener(ReactNativeModels.ReactDevToolsBindingsModel.Events.BackendExecutionContextDestroyed, this.onBackendExecutionContextDestroyed, this);
+  async ensureInitialized(): Promise<void> {
+    if (this.#initialized) {
+      return;
+    }
 
-    void this.initialize(rdtBindingsModel);
+    await this.#bindingsModel.enable()
+      .then(() => this.#handleBindingsModelInitializationCompleted())
+      .catch((error: Error) => this.#handleBindingsModelInitializationFailed(error));
   }
 
-  private async initialize(rdtBindingsModel: ReactNativeModels.ReactDevToolsBindingsModel.ReactDevToolsBindingsModel): Promise<void> {
-    return rdtBindingsModel.enable()
-      .then(() => this.onBindingsModelInitializationCompleted())
-      .catch((error: Error) => this.onBindingsModelInitializationFailed(error));
+  isInitialized(): boolean {
+    return this.#initialized;
   }
 
-  private onBindingsModelInitializationCompleted(): void {
-    const rdtBindingsModel = this.rdtBindingsModel;
-    if (!rdtBindingsModel) {
+  getBridge(): ReactDevToolsTypes.Bridge {
+    if (this.#bridge == null) {
+      throw new Error('Failed to get bridge from ReactDevToolsModel: bridge was null');
+    }
+
+    return this.#bridge;
+  }
+
+  getStore(): ReactDevToolsTypes.Store {
+    if (this.#store == null) {
+      throw new Error('Failed to get store from ReactDevToolsModel: store was null');
+    }
+
+    return this.#store;
+  }
+
+  #handleBindingsModelInitializationCompleted(): void {
+    const bindingsModel = this.#bindingsModel;
+    if (!bindingsModel) {
       throw new Error('Failed to initialize ReactDevToolsModel: ReactDevToolsBindingsModel was null');
     }
 
-    rdtBindingsModel.subscribeToDomainMessages(
+    bindingsModel.subscribeToDomainMessages(
       ReactDevToolsModel.FUSEBOX_BINDING_NAMESPACE,
-        message => this.onMessage(message as ReactDevToolsTypes.Message),
+        message => this.#handleMessage(message as ReactDevToolsTypes.Message),
     );
 
-    void rdtBindingsModel.initializeDomain(ReactDevToolsModel.FUSEBOX_BINDING_NAMESPACE)
-      .then(() => this.onDomainInitializationCompleted())
-      .catch((error: Error) => this.onDomainInitializationFailed(error));
+    void bindingsModel.initializeDomain(ReactDevToolsModel.FUSEBOX_BINDING_NAMESPACE)
+      .then(() => this.#handleDomainInitializationCompleted())
+      .catch((error: Error) => this.#handleDomainInitializationFailed(error));
+
+    this.#initialized = true;
```

Comment:
I believe this should be in `handleDomainInitializationCompleted`

Reply:
That's fair. I'm introducing an `#initializing` variable to avoid the promise execution (2 of them) race condition.